clang/test/Driver/sycl-cuda-arch-macro.cpp

```diff
@@ -1,9 +1,13 @@
 // Verify the __CUDA_ARCH__ macro has not been defined when offloading SYCL on NVPTX
 // RUN: %clangxx -E -dM -fsycl -fsycl-targets=nvptx64-nvidia-cuda -Xsycl-target-backend --offload-arch=sm_80 -nocudalib -fno-sycl-libspirv %s 2>&1 \
 // RUN: | FileCheck --check-prefix=CHECK-CUDA-ARCH-MACRO %s
+// Verify the __CUDA_ARCH__ macro has not been defined when offloading SYCL on AMDGPU
+// RUN: %clangxx -E -dM -fsycl -fsycl-targets=amdgcn-amd-amdhsa -Xsycl-target-backend --offload-arch=gfx90a -nogpulib -fno-sycl-libspirv %s 2>&1 \
```

Comment:
Is the behavior same in `CL` mode?

Reply:
Do you mean OpenCL by that? If so, that does not happen in CL mode. The definition of the `__CUDA_ARCH__` is dependent on whether we have Cuda device mode enabled and that only happened in the Cuda and HIP toolchains.

Reply:
I meant `clang_cl` executable.

Reply:
I see. Yeah, should be. I'll add follow ups testing clang cl in a new PR to address both NVPTX and AMDGPU.